### PR TITLE
fossil: 2.14 -> 2.15.1

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126232` at 6a2848b8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fossil</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/version-management/fossil/default.nix:31:3:
```
   |
31 |   buildInputs = [ zlib openssl readline sqlite which ed ]
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/version-management/fossil/default.nix:49:3:
```
   |
49 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/version-management/fossil/default.nix [hunk 1]
```diff
@@ -49,7 +51,7 @@ stdenv.mkDerivation rec {
     INSTALLDIR=$out/bin make install
 
     installManPage fossil.1
-    installShellCompletion --name fossil.bash tools/fossil-autocomplete.bash
+    installShellCompletion --name fossil.bbash tools/fossil-autocomplete.bash
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/fossil/default.nix:54]: You are sure about this change?

File: pkgs/applications/version-management/fossil/default.nix [hunk 2]
```diff
@@ -64,5 +66,6 @@ stdenv.mkDerivation rec {
     homepage = "http://www.fossil-scm.org/";
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/fossil/default.nix:66]:
```suggestion
    homepage = "https://www.fossil-scm.org/";
```

File: pkgs/applications/version-management/fossil/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/applications/version-management/fossil/default.nix:54]: Oops, must have been a typo.

**Pull request merged**